lib/emerald/parser.rb

```diff
@@ -6,17 +6,27 @@ def initialize(source)
 
     def parse
       ast = []
-      result = parse_node(@source)
+      parse_lists_and_nodes(@source, ast)
+    end
+
+    private
+
+    def parse_lists_and_nodes(source, ast)
+      result = parse_node(source)
       while result
         node, source = result
-        ast.push(node) if node
+        if node.is_a? List
```

Comment:
There's some code here that only happens if we're parsing a node. Perhaps it would be better if this happened in the `parse_list` method.